packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -4965,6 +4967,148 @@ void main() {
     expect(merged.constraints, overrideTheme.constraints);
   });
 
+  testWidgets('Prefix and suffix IconButtons inherit IconButtonTheme', (WidgetTester tester) async {
+    const IconData prefixIcon = Icons.person;
+    const IconData suffixIcon = Icons.search;
+    const Color backgroundColor = Color(0xffff0000);
+    const Color foregroundColor = Color(0xff00ff00);
+    final OutlinedBorder shape =RoundedRectangleBorder(
+      borderRadius: BorderRadius.circular(10.0),
+    );
+    final ButtonStyle iconButtonStyle = IconButton.styleFrom(
+      backgroundColor: backgroundColor,
+      foregroundColor: foregroundColor,
+      shape: shape,
+    );
+
+    await tester.pumpWidget(
+      IconButtonTheme(
+        data: IconButtonThemeData(style: iconButtonStyle),
+        child: buildInputDecorator(
+          decoration: InputDecoration(
+            prefixIcon: IconButton(
+              onPressed: () {},
+              icon: const Icon(prefixIcon),
+            ),
+            suffixIcon: IconButton(
+              onPressed: () {},
+              icon: const Icon(suffixIcon),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final Finder prefixIconMaterial = find.descendant(
+      of: find.widgetWithIcon(IconButton, prefixIcon),
+      matching: find.byType(Material),
+    );
+    Material material = tester.widget<Material>(prefixIconMaterial);
+    expect(material.color, backgroundColor);
+    expect(material.shape, iconButtonStyle.shape?.resolve(<WidgetState>{}));
+    final Finder suffixIconMaterial = find.descendant(
+      of: find.widgetWithIcon(IconButton, suffixIcon),
+      matching: find.byType(Material),
+    );
+    material = tester.widget<Material>(suffixIconMaterial);
+    expect(material.color, backgroundColor);
+    expect(material.shape, shape);
+
+    expect(getIconStyle(tester, prefixIcon)?.color, foregroundColor);
+    expect(getIconStyle(tester, suffixIcon)?.color, foregroundColor);
+  });
+
+  testWidgets('Prefix IconButton color respects IconButtonTheme foreground color states', (WidgetTester tester) async {
+    const IconData prefixIcon = Icons.person;
+    const Color IconErrorColor = Color(0xffff0000);
```

Comment:
Nit: This should be lower case right? iconErrorColor.

Reply:
ooops... good catch!